File: rb/features.rb
# TruthOrDare SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TruthOrDareFeatures
  def self.make_feature(name)
    case name
    when "base"
      TruthOrDareBaseFeature.new
    when "test"
      TruthOrDareTestFeature.new
    else
      TruthOrDareBaseFeature.new
    end
  end
end

File: py/features.py
# TruthOrDare SDK feature factory

from feature.base_feature import TruthOrDareBaseFeature
from feature.test_feature import TruthOrDareTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TruthOrDareBaseFeature(),
        "test": lambda: TruthOrDareTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
